animal.go

```diff
@@ -32,9 +32,11 @@ func GetUserYesOrNo(question string) string {
 	scanner := bufio.NewScanner(os.Stdin)
```

Comment:
Instead of `os.Stdin`, this should be the `io.Reader` parameter that's passed in to the function (see below).

Reply:
Fixed during mentoring session.